torch/_ops.py

```diff
@@ -1353,17 +1361,19 @@ def _refresh_packet(packet):
     packet._overload_names = overload_names
 
 
-class _PyOpNamespace(_OpNamespace):
```

Comment:
What is the purpose of this refactor? Is `_PyOpNamespace` single use, so this is more expressive or was there an issue with typing that this helps resolve?

Reply:
OK, there are several things in play here:

1. The base `_OpNamespace` class needs to annotate `__getattr__` with return type `OpOverloadPacket`, for maximum accuracy. This class, OTOH, returns `HigherOrderOperator`, which is an incompatible overload.  To avoid returning an unnecessary `Union` type, and since all but one method in the base was overridden anyway, I figured it was best to break out the class into something standalone.
2. This class was single use, so after breaking it out into its own class, I was able to remove a lot of unneeded complexity by just specializing it.
3. Once the class was appropriately specialized, it became obvious that it could be used once as an explicit member of `_Ops`, which simplified the typing for `_Ops.__getattr__`.

Reply:
Sounds reasonable to me.